ScopedStorage/app/src/main/java/com/samples/storage/mediastore/AddMediaFragment.kt

```diff
@@ -16,24 +16,51 @@
 
 package com.samples.storage.mediastore
 
+import android.app.Activity
+import android.content.ContentValues
+import android.content.Intent
+import android.net.Uri
+import android.os.Build
 import android.os.Bundle
+import android.provider.MediaStore
+import android.util.Log
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.activity.result.contract.ActivityResultContracts.TakePicture
 import androidx.fragment.app.Fragment
-import com.samples.storage.databinding.FragmentDemoBinding
+import androidx.fragment.app.viewModels
+import com.samples.storage.databinding.FragmentAddMediaBinding
 
 class AddMediaFragment : Fragment() {
-    private var _binding: FragmentDemoBinding? = null
+    private var _binding: FragmentAddMediaBinding? = null
     private val binding get() = _binding!!
+    private val viewModel: AddMediaViewModel by viewModels()
+
+    private val actionTakePicture = registerForActivityResult(TakePicture()) { success ->
+        if (!success) {
+            Log.d(tag, "Image taken FAIL: ${viewModel.temporaryMediaUri.value}")
```

Comment:
nit: maybe make `tag` a `const val TAG`?

Reply:
Why? It's a property from Fragment (which avoids having to write a constant)

Reply:
Ah, I didn't notice that. sgtm then.